### ISSUE TITLE
[Vue] Don't format custom blocks

### ISSUE THREAD

**Issue by coxy** [prettier/prettier#5448]
**Prettier 1.15.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmpaEAqc2+8AzmXgE5xlQQxkBmENhgLwAJmQCecXgDo5pChVQB6aqRWZcBOCSjqAlgEYAHFF0UESBYuhwkZAPJQ41ijADuEezU-rlR0111JjA2fSwYXRxpMgAFDhgYfTg2MgBefkFhXjTCMmBJCABXNhkAKyYACgByGAALYoBzOuqASjIAXz8QsIigqFQmGAlo3Rk8KHECsCYmexgICDIAQn1sQRgJ3i6B5SGRnRAAGhAICP1oJmRQdjYId1j2BCuUPAA3CH1RY5AAIzY8GAANbSADKWEB+igjWQMDYRTgJzqMAwOAA6nV9MwIWA4KDnlj9G8sRJkOBZj8oUwUjB4nhGsJkHw8DhqScKgAPABCAOBYLwGDgABkoXAmSy2SBOaCoY1ogBFIo8MVIZmsxEgCFsalsMlvBE-LBhWBor71ZDGAAMJyNEGpaIBWDJRrgOreYpOHAAjkV9Bw6Qy8OL1SdqRh9LD4RqmLKFUr4MHJZtfqbROakAAmE5wvD6HCygDCEAwjJQ3GcPyK1JoeF+LzV1I6HSAA)
```sh
--parser vue
```

**Input:**
```vue
<template>
  <p>Templates are not formatted yet ...
    </p>
</template>

<i18n>
  en:
    one: One
    two: Two
</i18n>

<script>
let Prettier = format => { your.js('though') }
</script>

<style>
.and { css: too !important }
</style>
```

**Output:**
```vue
<template>
  <p>Templates are not formatted yet ...</p>
</template>

<i18n> en: one: One two: Two </i18n>

<script>
let Prettier = format => {
  your.js("though");
};
</script>

<style>
.and {
  css: too !important;
}
</style>

```

**Expected behavior:**
In my project I'm using https://kazupon.github.io/vue-i18n/guide/sfc.html#yaml-loading with a custom `<i18n>` block in the component file for translations. Here the format is yaml and prettier is formatting it incorrectly (I assume it treats it as plaintext not yaml).

More generally, vue-loader has support for any type of custom block https://vue-loader.vuejs.org/guide/custom-blocks.html#example - so my suggestion would be to ignore these blocks unless their type can be accurately identified. 

Perhaps ideally something like:
```
<myBlock type="text/yaml">
  yaml: Some text
</myBlock>
```
=> Prettier recognises the mime type and formats as yaml
```
<myBlock>
  Some text
      With a weird format
</myBlock>
```
=> Prettier doesn't know what mime type this block is and so ignores it and leaves formatting exactly as given in input

**Comment by ikatyang** [comment]
What's the definition of custom block in Vue? A top-level element that is not `template`, `style`, `script`?

**Comment by ldrick** [comment]
In my case <i18n></i18n> contains JSON and this breaks, due to the mentioned reformatting. I am using this project too
https://github.com/kazupon/vue-i18n to provide multiple languages in my components.

**Comment by coxy** [comment]
> What's the definition of custom block in Vue? A top-level element that is not `template`, `style`, `script`?

@ikatyang Yes, as far as I'm aware - though this could be worth clarifying with @yyx990803 or another maintainer of vue-loader. I also noticed that vue-loader uses the `lang` attribute for determining block type, rather than `type` so I have updated the issue. 

I think the safest/first thing to do here however, is probably just to ignore all custom blocks. Parsing/formatting based on block type could be a later enhancement.
